packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,692 @@
+
+  /**
+   * Sets what data to be rendered (file icon only, oneLine, twoLines threeLines).
+   * Default is 'threeLines'.
+   *
+   * @type {ViewType}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'item-view',
+    converter: value => {
+      if (!value || value.length === 0) {
+        return ViewType.threelines;
+      }
+
+      value = value.toLowerCase();
+
+      if (typeof ViewType[value] === 'undefined') {
+        return ViewType.threelines;
+      } else {
+        return ViewType[value];
+      }
+    }
+  })
+  public itemView: ViewType;
+
+  /**
+   * allows developer to provide file type to filter the list
+   * can be docx
+   *
+   * @type {string[]}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'file-extensions',
+    converter: (value, type) => {
+      return value.split(',').map(v => v.trim());
+    }
+  })
+  public get fileExtensions(): string[] {
+    return this._fileExtensions;
+  }
+  public set fileExtensions(value: string[]) {
+    if (arraysAreEqual(this._fileExtensions, value)) {
+      return;
+    }
+
+    this._fileExtensions = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * A number value to indicate the number of more files to load when show more button is clicked
+   * @type {number}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'page-size',
+    type: Number
+  })
+  public pageSize: number;
+
+  /**
+   * The selected item
+   *
+   * @readonly
+   * @memberof MgtFileList
+   * @type {MicrosoftGraph.DriveItem}
+   */
+  public get selectedItem() {
+    return this._selectedItem;
+  }
+
+  @internalProperty()
+  private _selectedItem: DriveItem;
+
+  private _fileListQuery: string;
+  private _fileQueries: string[];
+  private _files: DriveItem[];
+  private _siteId: string;
+  private _itemId: string;
+  private _driveId: string;
+  private _itemPath: string;
+  private _groupId: string;
+  private _insightType: OfficeGraphInsightString;
+  private _fileExtensions: string[];
+  private _userId: string;
+  private pageIterator: GraphPageIterator<DriveItem>;
+
+  constructor() {
+    super();
+
+    this.clearState();
+    this.pageSize = 10;
+    this._selectedItem = null;
+    this.itemView = ViewType.twolines;
+  }
+
+  /**
+   * Synchronizes property values when attributes change.
+   *
+   * @param {*} name
+   * @param {*} oldValue
+   * @param {*} newValue
+   * @memberof MgtPersonCard
+   */
+  public attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    super.attributeChangedCallback(name, oldValue, newValue);
+
+    if (oldValue === newValue) {
+      return;
+    }
+
+    switch (name) {
+      case 'file-list-query':
+      case 'file-queries':
+      case 'site-id':
+      case 'drive-id':
+      case 'group-id':
+      case 'item-id':
+      case 'item-path':
+      case 'user-id':
+      case 'insight-type':
+        this.clearState;
+        this.requestStateUpdate();
+        break;
+    }
+  }
```

Comment:
```suggestion
  protected requestStateUpdate(force?: boolean) {
    this.clearState();
    return super.requestStateUpdate(force);
  }
```

This should work to handle clearing the state when you need to request a state update. The `attributeChangedCallback` doesn't always get called when a property is changed because the underling attribute for a property is not always updated.

@shweaver-MSFT, what do you think about updating the baseComponent `requestStateUpdate` to call `clearState` by default - it looks like most components already overwrite this method to clearState anyway.

Reply:
Sounds good to me, but let's save that for a separate PR :)